src/main/java/com/xmage/launcher/XMageLauncher.java

```diff
@@ -540,10 +504,31 @@ private boolean getConfig() {
         return false;
     }
 
+    private boolean checkJavaFX() {
+        try {
+            Class.forName("javafx.application.Platform");
+            return true;
+        } catch( ClassNotFoundException e ) {
+            return false;
+        }
+    }
+
```

Comment:
I think this functionality is not needed (at least for the beta) as the Javafx dependancy has been removed. If it gets re-introduced in the future, it will be added to the client's pom.xml so that the project deals with its dependancies rather than hoisting that onto the user.

Reply:
should I remove it entirely? or should I just comment line 509 for future reintroduction?

Reply:
Leave it in there for now. I am not sure if we will need it or not, I have not yet looked into how to deal with javafx in the future if we switch to it over swing.

- [ ] comment out line 509
- [ ] Write a doctoring for `checkJava()`
- [ ] comment out calls to `checkJava()`

The last two will make it clearer what's going for anyone who comes by to look at it, and to work with it in the future.

Reply:
As third option, I think you mean to comment out calls to `checkJavaFx()` inside `checkJava()`, right?

Reply:
I originally meant checkJava, but it does seem to be doing something even if its set to not use system Java. I'll change the bullet point

Reply:
If JavaFX is not needed I'd go for option 1 + option 3 which is always useful

Reply:
2 and 3 are part of the same thing. For 2 I’m asking for a docstring overall what the function is doing, and for 3 I’m asking for the same information spread throughout the function so that anyone who needs to edit the function knows what each block of code is doing without having to read through it. I think both are needed (though 2 is just a summary of the points of 3).

Reply:
Done